File: .github/scripts/dependency/Change.py
"""A dependency update selection."""

from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class Change:
    """The current and selected value for one dependency."""

    name: str
    current: str
    latest: str

    @property
    def changed(self) -> bool:
        return self.current != self.latest

File: .github/scripts/src/Dependency/Change.php
<?php

declare(strict_types=1);

namespace DockerBase\Dependency;

final readonly class Change
{
    public function __construct(
        public string $name,
        public string $current,
        public string $latest,
        public ?string $currentReference = null,
        public ?string $latestReference = null,
    ) {
    }

    public function changed(): bool
    {
        return $this->current !== $this->latest
            || $this->currentReference !== $this->latestReference;
    }
}
